### PR TITLE
Implement start_function section

### PR DESCRIPTION
**Pull request by commonlisp**
Decode start_function section and run upon instantiateModule()

### CODE REVIEW

File: lib/Runtime/Base/ScriptContext.cpp [hunk 1]
```diff
@@ -2051,6 +2051,14 @@ namespace Js
                 indirectFunctionTables[sigId][i] = localModuleFunctions[funcIndex];
             }
 
+            uint32 startFunc = wasmModule->info->GetStartFunction();
+            if (startFunc != -1)
+            {
+                Js::Var v = localModuleFunctions[startFunc];
+                PropertyRecord const * mainProp = nullptr;
+                GetOrAddPropertyRecord(_u("main"), lstrlen(_u("main")), &mainProp);
```
Comment (MikeHolman) [inline lib/Runtime/Base/ScriptContext.cpp]: is this true? I don't think we should be adding this onto export object. we need only call it.

File: lib/Runtime/Library/WasmLibrary.cpp [hunk 2]
```diff
@@ -59,6 +59,14 @@ namespace Js
             );
         END_LEAVE_SCRIPT_INTERNAL(scriptContext)
 
+        Js::PropertyRecord const * mainProp = nullptr;
+        scriptContext->GetOrAddPropertyRecord(_u("main"), lstrlen(_u("main")), &mainProp);
+        Js::Var mainFunc = Js::JavascriptOperators::OP_GetProperty(exportObject, mainProp->GetPropertyId(), scriptContext);
+        Js::ScriptFunction* f = Js::AsmJsScriptFunction::FromVar(mainFunc);
+        Js::CallInfo info(Js::CallFlags_New, 1);
+        Js::Arguments arg0(info, &mainFunc);
+        Js::JavascriptFunction::CallFunction<true>(f, f->GetEntryPoint(), arg0);
```
Comment (MikeHolman) [inline lib/Runtime/Library/WasmLibrary.cpp]: seems like this is now requiring an export function. I think we will want start function to be an optional out parameter of LoadWasmScript, and if it is not null only then we call it.

Comment (commonlisp) [inline lib/Runtime/Library/WasmLibrary.cpp]: Good point. Revised based on that and rebased. Thanks.

**Comment by commonlisp** [comment]
@dotnet-bot test this please

**Comment by commonlisp** [comment]
@dotnet-bot test this please

**Comment by commonlisp** [comment]
@dotnet-bot test this please

**Comment by commonlisp** [comment]
@dotnet-bot test this please

**Comment by commonlisp** [comment]
@dotnet-bot test this please

**Comment by commonlisp** [comment]
@MikeHolman Any additional feedback on this one? Thanks!

**Comment by MikeHolman** [comment]
lgtm

File: lib/WasmReader/ModuleInfo.h [hunk 3]
```diff
@@ -76,6 +78,8 @@ namespace Wasm
         uint32 m_importCount;
         uint32 m_datasegCount;
 
+        uint32 m_startFun;
```
Comment (ThomsonTan) [inline lib/WasmReader/ModuleInfo.h]: m_startFunc may be better?

Comment (commonlisp) [inline lib/WasmReader/ModuleInfo.h]: Sure. Thanks!

**Pull request merged**